### DISCUSSION TITLE
Alpine.js intersect plugin broken after update of Livewire component

### DISCUSSION THREAD

**Discussion by gdebrauwer** [livewire/livewire#4480] [7. Bugs (archived)]
**Livewire:** v2.8.2
**Alpine.js:** v3.7.1

**Description**

When a Livewire component is updated, then the intersect plugin thinks the item is not in viewport while it is visible in the viewport.

**Example to recreate bug**

In this example, I create a simple Livewire component has 2 HTML elements:
- a div that uses the intersect plugin. it is always visible in the viewport, as there is nothing else on the page
- a button that simply refreshes the Livewire component.

On the initial page load, the div correctly shows the text 'in viewport'. But if you click the button, the text will change to 'not in viewport' when Livewire has refreshed the component. That is not correct, as the div is still visible in the viewport.

```js
import Alpine from 'alpinejs';
import intersect from '@alpinejs/intersect';

window.Alpine = Alpine;

Alpine.plugin(intersect);

Alpine.start();
```

```php
class IntersectAlpineIssue extends Component
{
    public function render()
    {
        return view('livewire.intersect-alpine-issue');
    }
}
```

```blade
<div>
    <div
        x-data="{ inViewport: false }"
        x-intersect:enter="inViewport = true"
        x-intersect:leave="inViewport = false"
        x-effect="console.log(inViewport)"
        x-text="inViewport ? 'in viewport': 'not in viewport'"
    >
    </div>

    <button type="button" wire:click="$refresh">button</button>
</div>
```

```php
// in routes/web.php
Route::get('intersect-issue', IntersectAlpineIssue::class);
```

**Comment by joshhanley** [comment]
@gdebrauwer thanks for reporting! Hmm yeah it's possible that the plugin doesn't have Livewire support yet. Have you posted this in the Alpine repo?

**Reply by gdebrauwer** [reply]
I only reported it in this repository, as I was not sure where to post it 😅 I now created an issue in the Alpine.js repository as well: https://github.com/alpinejs/alpine/discussions/2576

**Reply by joshhanley** [reply]
@gdebrauwer yeah fair enough. As the code for the plugin is in that repository, that's why I suggested to post it there 🙂

**Comment by cbaconnier** [comment]
In my case, adding `wire:ignore` fixed the issue.

I was using this to lazy-load images but refreshing the content was not triggering the `x-intersect` and the image was removed from src, but this works very well !

```blade
@props(['src'])
<img
    wire:ignore
    x-data="{ src: {{ Js::from($src) }} }"
    x-intersect:enter.once="$root.src = src"
    {{ $attributes }}
/>
```

**Comment by kaiserkais** [comment]
guys who found the solution for this problem

**Comment by gdebrauwer** [comment]
Failing test PR created: https://github.com/livewire/livewire/pull/7880
